### PR TITLE
Add string query manual source

### PR DESCRIPTION
**Pull request by akremstudy**

### CODE REVIEW

File: src/telliot_feeds/sources/manual_sources/string_query_manual_source.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+
+
+class StringQueryManualSource(DataSource[Optional[str]]):
+    async def fetch_new_datapoint(self) -> OptionalDataPoint[str]:
+        text = None
+        msg = "Type your string query response:"
+        print(f"{msg}\n")
```
Comment (oraclown) [inline src/telliot_feeds/sources/manual_sources/string_query_manual_source.py:12]: Get rid of var `msg` and just print the whole message

File: tests/sources/test_manual_sources.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+async def test_manual_spot_price_good_input(
+    monkeypatch,
+):
+    monkeypatch.setattr("builtins.input", lambda: "1234")
+    result, _ = await SpotPriceManualSource().fetch_new_datapoint()
+    assert result == 1234
+
+
+@pytest.mark.asyncio
+async def test_manual_spot_price_bad_input(capsys):
```
Comment (oraclown) [inline tests/sources/test_manual_sources.py:22]: I think we should have separate files for each manual source tests. They could get to be more than just one bad input, one good input test

File: tests/sources/test_manual_sources.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        captured_output = capsys.readouterr()
+        assert expected in captured_output.out.strip()
+        assert type(result) is float
+        assert result == 5678
+
+
+@pytest.mark.asyncio
+async def test_string_query_good_input(capsys):
+    inpt = "hello"
+    with mock.patch("builtins.input", side_effect=[inpt, ""]):
```
Comment (oraclown) [inline tests/sources/test_manual_sources.py:35]: If you have a single variable that is only used one later, you often don't need to instantiate it, so in this example just get rid of `inpt` & pass in the string `hello` directly to the list. It reads cleaner is less for the reader to keep track of

File: tests/sources/test_manual_sources.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        result, _ = await StringQueryManualSource().fetch_new_datapoint()
+        assert isinstance(result, str)
+        assert result == "hello"
+
+
+@pytest.mark.asyncio
+async def test_string_query_bad_input(capsys):
```
Comment (oraclown) [inline tests/sources/test_manual_sources.py:42]: Add a test for the user just pressing [ENTER], so the input would be an empty string

File: src/telliot_feeds/sources/manual_sources/string_query_manual_source.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+        while text is None:
+            usr_inpt = input()
+
+            # not sure what to catch here but possible bytes that could be passed via something other than the cli
+            # since everything will be interepreted to string when passed through the cli.
+            if len(usr_inpt) > 1 and isinstance(usr_inpt, str):
```
Comment (oraclown) [inline src/telliot_feeds/sources/manual_sources/string_query_manual_source.py:19]: Any string goes, even if they want to put `"b'blah'"` on chain. Maybe just a message to let them know they're going to report an empty string if that's the case. Anything else just preview the string they'll be putting on chain before

File: src/telliot_feeds/reporters/reporter_autopay_utils.py [hunk 6]
```diff
@@ -311,21 +311,26 @@ async def get_feed_tip(query: bytes, autopay: TellorFlexAutopayContract) -> Opti
         msg = "can't suggest feed, autopay contract not connected"
         error_status(note=msg, log=logger.critical)
         return None
-    try:
+
+    if query in CATALOG_QUERY_IDS:
```
Comment (oraclown) [inline src/telliot_feeds/reporters/reporter_autopay_utils.py:315]: Why's it named `query`? Isn't the key for this dictionary the stringified `queryId`?

File: src/telliot_feeds/reporters/reporter_autopay_utils.py [hunk 7]
```diff
@@ -311,21 +311,26 @@ async def get_feed_tip(query: bytes, autopay: TellorFlexAutopayContract) -> Opti
...
-        if qtype_name not in Registry.registry:
-            logger.warning(f"Unsupported query type: {qtype_name}")
+    else:
+        try:
+            qtype_name, _ = decode_single("(string,bytes)", query)
+            if qtype_name not in Registry.registry:
+                logger.warning(f"Unsupported query type: {qtype_name}")
+                return None
+            else:
+                query = Web3.keccak(query)
```
Comment (oraclown) [inline src/telliot_feeds/reporters/reporter_autopay_utils.py:324]: This one seems like it should def be named `query_id`, no? I only nitpick on the naming bc it'll be confusing for future readers (like me lol)

File: src/telliot_feeds/reporters/reporter_autopay_utils.py [hunk 8]
```diff
@@ -311,21 +311,26 @@ async def get_feed_tip(query: bytes, autopay: TellorFlexAutopayContract) -> Opti
...
+                CATALOG_QUERY_IDS[query] = query.hex() + qtype_name.lower()
```
Comment (oraclown) [inline src/telliot_feeds/reporters/reporter_autopay_utils.py:325]: This looks wrong. You're concatenating the string queryId w/ the query type string? Why? Maybe add a comment to help explain..

File: src/telliot_feeds/reporters/reporter_autopay_utils.py [hunk 9]
```diff
@@ -311,21 +311,26 @@ async def get_feed_tip(query: bytes, autopay: TellorFlexAutopayContract) -> Opti
...
+                single_query = {query: CATALOG_QUERY_IDS[query]}
+        except OverflowError:
```
Comment (oraclown) [inline src/telliot_feeds/reporters/reporter_autopay_utils.py:327]: Your specific exceptions should wrap only the line that would generate the error, so put after line 319 I think

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks for adding these @akremstudy

File: src/telliot_feeds/reporters/reporter_autopay_utils.py (hunk 6 above)
Comment (akremstudy) [reply src/telliot_feeds/reporters/reporter_autopay_utils.py:315]: there is a comment about this here:  https://github.com/tellor-io/telliot-feeds/blob/10099d7c4d9ffbc2c5831c0f399bfa2cc9b5d761/src/telliot_feeds/reporters/reporter_autopay_utils.py#L306

File: src/telliot_feeds/reporters/reporter_autopay_utils.py (hunk 7 above)
Comment (akremstudy) [reply src/telliot_feeds/reporters/reporter_autopay_utils.py:324]: yes you're right

File: src/telliot_feeds/reporters/reporter_autopay_utils.py (hunk 9 above)
Comment (akremstudy) [reply src/telliot_feeds/reporters/reporter_autopay_utils.py:327]: will do

**Comment by akremstudy** [comment]
closes #287

**Review by oraclown — APPROVED** [review]
Thanks @akremstudy !
